### ISSUE TITLE
NPM Release and Typescript Support

### ISSUE THREAD

**Issue by michaelcmelton** [freshfork/p5.EasyCam#13]
Is there current Typescript Support and NPM packages for this library?

**Comment by jwdunn1** [comment]
Not yet. Perhaps NPM packaging sooner than Typescript support. I'll mark this as an enhancement.

**Comment by michaelcmelton** [comment]
Are you open to contributions? If so, I'd be glad to try and help with these things when I've got some free time.

**Comment by jwdunn1** [comment]
Yes, this library is open to contributions! When you get the opportunity, feel free to fork, update, test, and create a pull request.

**Comment by michaelcmelton** [comment]
@jwdunn1  Do you have any contact info I can reach you at? I want to run a few things by you before I begin really digging in.

**Comment by wuyudi** [comment]
https://github.com/wuyudi/p5.EasyCam/blob/try2ts/dist/p5.easycam.d.ts

I spent some time to adopt it into ts and generate a .d.ts file. Hope it may help.

Some types are not clear to me, so I write `any` instead.

I don't know whether this can be a PR (ts to js is a huge difference).

**Comment by jwdunn1** [comment]
@wuyudi Excellent! Thank you for the time spent on this. If you want to create a PR, please rename or drop the **dist** folder - it may be misunderstood as a distribution folder.

**Comment by jwdunn1** [comment]
NPM package [published](https://www.npmjs.com/package/p5.easycam).
[ It took more than one year to resolve the request for control of the name "p5.easycam" on npm! ]

**Comment by jwdunn1** [comment]
The TypeScript version of the code is v1.2.0. This is not aligned with the current release version 1.2.2.
